app/gui2/src/util/ast/extended.ts

```diff
@@ -93,6 +108,20 @@ export class AstExtended<T extends Tree | Token = Tree | Token, HasIdMap extends
     return new AstExtended(mapper(this.inner), this.ctx)
   }
 
+  mapIter<T2 extends Tree | Token>(
+    mapper: (t: T) => Iterable<T2>,
+  ): Iterable<AstExtended<T2, HasIdMap>> {
+    return [...mapper(this.inner)].map((m) => new AstExtended(m, this.ctx))
+  }
+
+  tryMapIter<T2 extends Tree | Token>(
+    mapper: (t: T) => Iterable<Opt<T2>>,
+  ): Iterable<AstExtended<T2, HasIdMap> | undefined> {
+    return [...mapper(this.inner)].map((m) =>
+      m != null ? new AstExtended(m, this.ctx) : undefined,
+    )
+  }
```

Comment:
Needed for handling things like `OperatorBlockApplication.expressions`, which returns Iterable. Utilized in `opr.ts` and `input.ts`.

Reply:
should the returned `Iterable` have `undefined`? (is it necessary for indices to match in the input and output iterables? the alternative would be to `.flatMap` + `m != null ? [new AstExtended(...)] : []`)

Reply:
Yeah, I’m not sure, I need to think about it a bit. Thanks for suggestion.

Reply:
I think `mapIter` has the same functionality as `visit`. Perhaps the latter is a bit more efficient as it does not create intermediate arrays.

Reply:
Firstly, `Iterable` must return nullable values as well, it is needed for correct handling of `OperatorLine`s.
Secondly, `mapIter` is indeed similar to `visit`, but the difference is crucial, as in some places we can’t use generators easily, at least without heavy refactoring. I made a few attempts to reuse `visit` here, but didn’t find anything particularly good.